python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Dict, Mapping
+
+from dagster import AssetKey
+
+from ..asset_utils import default_asset_key_fn, output_name_fn
+
+
+@dataclass
+class DbtManifest:
```

Comment:
I think we should provide a static constructor for this object, so that people can do `DbtManifest.from_path`, or `DbtManifest.from_dict`. This saves them from having to do the whole with open(...): json.loads(...) stuff on their own, and also allows us (in the future) to take advantage of the `WriteableManifest.from_file` or whatever it's called in dbt land